blazor/datagrid/sorting.md

```diff
@@ -264,23 +398,649 @@ In the following code example, custom SortComparer class was defined in the Cust
+
+    protected override void OnInitialized()
+    {
+        GridData = OrderData.GetAllRecords();
+    }
+
+    List<Columns> LocalData = new List<Columns>
+    {
+        new Columns() { ID= "OrderID", Value= "OrderID" },
+        new Columns() { ID= "CustomerID", Value= "CustomerID" },
+        new Columns() { ID= "Freight", Value= "Freight" },
+    };
+
+    List<Direction> LocalData1 = new List<Direction>
+    {
+        new Direction() { ID= "Ascending", Value= "Ascending" },
+        new Direction() { ID= "Descending", Value= "Descending" },
+
+    };
+
+    public class Columns
+    {
+        public string ID { get; set; }
+        public string Value { get; set; }
+    }
+
+    public class Direction
+    {
+        public string ID { get; set; }
+        public string Value { get; set; }
+    }
+
+    public void AddsortColumn()
+    {
+        Grid.SortColumnAsync(DropDownValue, DropDownDirection, true);
+    }
+}
+{% endhighlight %}
+{% highlight c# tabtitle="OrderData.cs" %}
+ public class OrderData
+    {
+        public static List<OrderData> Orders = new List<OrderData>();
+        
+        public OrderData()
+        {
+
+        }
+        public OrderData(int? OrderID,string CustomerID, double? Freight, string ShipName)
+        {
+           this.OrderID = OrderID;    
+           this.CustomerID = CustomerID;
+            this.Freight = Freight;
+            this.ShipName = ShipName;             
+        }
+
+        public static List<OrderData> GetAllRecords()
+        {
+            if (Orders.Count() == 0)
+            {
+                int code = 10;
+                for (int i = 1; i < 2; i++)
+                {
+                    Orders.Add(new OrderData(10248, "VINET", 3.25, "Vins et alcools Chevali"));
+                    Orders.Add(new OrderData(10249, "TOMSP", 22.98, "Toms Spezialitäten"));
+                    Orders.Add(new OrderData(10250, "HANAR", 140.51, "Hanari Carnes"));
+                    Orders.Add(new OrderData(10251, "VICTE", 65.83, "Victuailles en stock"));
+                    Orders.Add(new OrderData(10252, "SUPRD", 58.17, "Suprêmes délices"));
+                    Orders.Add(new OrderData(10253, "HANAR", 81.91, "Hanari Carnes"));
+                    Orders.Add(new OrderData(10254, "CHOPS", 3.05, "Chop-suey Chinese"));
+                    Orders.Add(new OrderData(10255, "RICSU", 55.09, "Richter Supermarkt"));
+                    Orders.Add(new OrderData(10256, "WELLI", 48.29, "Wellington Import"));
+                    code += 5;
+                }
+            }
+            return Orders;
+        }
+
+        public int? OrderID { get; set; }
+        public string CustomerID { get; set; }
+        public string ShipName { get; set; }
+        public double? Freight { get; set; }
+    }
+{% endhighlight %}
+{% endtabs %}
+
+{% previewsample "https://blazorplayground.syncfusion.com/embed/LtBgsWDKBXuYpcUi?appbar=false&editor=false&result=true&errorlist=false&theme=bootstrap5" %}
+
```

Comment:
ensure remoteSortColumn method available or not

Reply:
Modified